src/sentry/api/serializers/models/incident.py

```diff
@@ -54,13 +55,29 @@ def serialize(self, obj, attrs, user):
             'dateDetected': obj.date_detected,
             'dateAdded': obj.date_added,
             'dateClosed': obj.date_closed,
+            'hasSeen': attrs['has_seen'],
             'eventStats': serializer.serialize(attrs['event_stats']),
             'totalEvents': aggregates['count'],
             'uniqueUsers': aggregates['unique_users'],
         }
 
 
 class DetailedIncidentSerializer(IncidentSerializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        results = super(DetailedIncidentSerializer, self).get_attrs(item_list, user)
+        seen_incidents = dict(
+            IncidentSeen.objects.filter(
+                incident__in=item_list
+            ).select_related('user').values_list('incident_id', 'last_seen')
+        )
+
+        for incident in item_list:
+            active_date = incident.date_started
+            results[incident]['has_seen'] = seen_incidents.get(
```

Comment:
Is this just to check if an incident has been seen at all?

Reply:
Yeah this is to check that it has been seen by the user (this was copied from the group seen logic)

Reply:
Since we're already fetching all `IncidentSeen` rows for each incident in serialize, should we just do this  there instead?